File: CS/ReportLoad/DataHelper.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraScheduler;
using System.IO;

namespace ReportLoad {
    class DataHelper {
        public const string aptDataResourceName = "ReportLoad.Data.appointments.xml";
        public const string resDataResourceName = "ReportLoad.Data.resources.xml";
        public const string templateName = "ReportLoad.Templates.WeeklyStyle.repx";


        public static Stream GetTemplate() {
            return System.Reflection.Assembly.GetExecutingAssembly().GetManifestResourceStream(templateName);
        
        }
        
        public static void FillStorageData(SchedulerStorage storage) {
            FillStorageCollection(storage.Resources.Items, resDataResourceName);
            FillStorageCollection(storage.Appointments.Items, aptDataResourceName);
        }

        static void FillStorageCollection(AppointmentCollection c, string resourceName) {
            using(Stream stream = GetResourceStream(resourceName)) {
                c.ReadXml(stream);
                stream.Close();
            }
        }
        static void FillStorageCollection(ResourceCollection c, string resourceName) {
            using(Stream stream = GetResourceStream(resourceName)) {
                c.ReadXml(stream);
                stream.Close();
            }
        }

        static Stream GetResourceStream(string resourceName) {
            return System.Reflection.Assembly.GetExecutingAssembly().GetManifestResourceStream(resourceName);
        }




    }
}

File: VB/ReportLoad/DataHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.XtraScheduler
Imports System.IO

Namespace ReportLoad
	Friend Class DataHelper
		Public Const aptDataResourceName As String = "ReportLoad.Data.appointments.xml"
		Public Const resDataResourceName As String = "ReportLoad.Data.resources.xml"
		Public Const templateName As String = "ReportLoad.Templates.WeeklyStyle.repx"


		Public Shared Function GetTemplate() As Stream
			Return System.Reflection.Assembly.GetExecutingAssembly().GetManifestResourceStream(templateName)

		End Function

		Public Shared Sub FillStorageData(ByVal storage As SchedulerStorage)
			FillStorageCollection(storage.Resources.Items, resDataResourceName)
			FillStorageCollection(storage.Appointments.Items, aptDataResourceName)
		End Sub

		Private Shared Sub FillStorageCollection(ByVal c As AppointmentCollection, ByVal resourceName As String)
			Using stream As Stream = GetResourceStream(resourceName)
				c.ReadXml(stream)
				stream.Close()
			End Using
		End Sub
		Private Shared Sub FillStorageCollection(ByVal c As ResourceCollection, ByVal resourceName As String)
			Using stream As Stream = GetResourceStream(resourceName)
				c.ReadXml(stream)
				stream.Close()
			End Using
		End Sub

		Private Shared Function GetResourceStream(ByVal resourceName As String) As Stream
			Return System.Reflection.Assembly.GetExecutingAssembly().GetManifestResourceStream(resourceName)
		End Function




	End Class
End Namespace
